libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -3720,6 +3727,71 @@ string YulUtilFunctions::arrayConversionFunction(ArrayType const& _from, ArrayTy
 	});
 }
 
+string YulUtilFunctions::inlineArrayConversionFunction(InlineArrayType const& _from, ArrayType const& _to)
+{
+	if (_to.dataStoredIn(DataLocation::CallData))
+		solAssert(false);
+
+	if (!_to.isDynamicallySized())
+		solAssert(_to.length() == _from.components().size());
+
+	// TODO
+	// Other cases are done explicitly in LValue::storeValue, and only possible by assignment.
+	/*if (_to.location() == DataLocation::Storage)
+		solAssert(
+			(_to.isPointer() || (_from.isByteArrayOrString() && _to.isByteArrayOrString())) &&
+			_from.location() == DataLocation::Storage,
+			"Invalid conversion to storage type."
+		);
+	*/
+	string functionName =
+		"convert_inline_array_" +
+		_from.identifier() +
+		"_to_" +
+		_to.identifier();
+
+		vector<map<string, string>> memberSetValues;
+				unsigned stackItemIndex = 0;
+				for (auto&& [index, type]: _from.components() | ranges::views::enumerate)
+				{
+					memberSetValues.emplace_back();
+
+					memberSetValues.back()["setMember"] = Whiskers(R"(
+						let <memberValues> := <conversionFunction>(<value>)
+						<writeToMemory>(add(mpos, <offset>), <memberValues>)
```

Comment:
```suggestion
					{
						let <memberValues> := <conversionFunction>(<value>)
						<writeToMemory>(add(mpos, <offset>), <memberValues>)
					}
```
Maybe it's nicer (if it actually works) to add a small scope for these? Then you might be able to drop the ``stackItemIndex`` thing.

Reply:
Ah, no, you need that for ``value``, resp. for ``var_*`` anyways... then nevermind :-).

Reply:
But wait... will the index you need for the ``var_*`` things always increase by the same amount as you have ``memberValues_*`` things?

My guess would be that for ``memberValues`` you need ``_to.baseType()->stackItems().size()`` slots each time instead, wouldn't you? And then just adding scopes and keeping them local each time might be easier.

Also the ``writeToMemoryFunction`` should be on the ``*_to.baseType()`` type instead, shouldn't it?

I.e. the conversion function each time takes ``<value>`` of type ``type`` and converts it to ``_to.baseType()``, so ``<memberValues>`` should have type ``_to.baseType()`` and consequently ``_to.baseType()->stackItems().size()``-many slots and should be written to memory using ``writeToMemoryFunction(*_to.baseType())``.

Reply:
Thank you for your remark and clarification! Code updated accordingly.